src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java

```diff
@@ -29,146 +32,252 @@
-				new BasicSSHUserPrivateKey.DirectEntryPrivateKeySource(
-						new String(Base64.decode(sshKeyData), StandardCharsets.UTF_8)),
-				null, secretName);
-	}
-
-	private static Credentials newUsernamePasswordCredentials(String secretName, String usernameData,
-			String passwordData) {
-		return new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, secretName, secretName,
-				new String(Base64.decode(usernameData), StandardCharsets.UTF_8),
-				new String(Base64.decode(passwordData), StandardCharsets.UTF_8));
-	}
-
-	/**
-	 * Does our configuration have credentials?
-	 * 
-	 * @return true if found.
-	 */
-	public static boolean hasCredentials() {
-		return !StringUtils.isEmpty(getAuthenticatedOpenShiftClient().getConfiguration().getOauthToken());
-	}
+    private final static Logger logger = Logger
+            .getLogger(CredentialsUtils.class.getName());
+    
+    public static synchronized Secret getSourceCredentials(
+            BuildConfig buildConfig) {
+        if (buildConfig.getSpec() != null
+                && buildConfig.getSpec().getSource() != null
+                && buildConfig.getSpec().getSource().getSourceSecret() != null
+                && !buildConfig.getSpec().getSource().getSourceSecret()
+                        .getName().isEmpty()) {
+            Secret sourceSecret = getAuthenticatedOpenShiftClient()
+                    .secrets()
+                    .inNamespace(buildConfig.getMetadata().getNamespace())
+                    .withName(
+                            buildConfig.getSpec().getSource().getSourceSecret()
+                                    .getName()).get();
+            return sourceSecret;
+        }
+        return null;
+    }
+
+    public static synchronized String updateSourceCredentials(
+            BuildConfig buildConfig) throws IOException {
+        Secret sourceSecret = getSourceCredentials(buildConfig);
+        if (sourceSecret != null) {
+            return upsertCredential(sourceSecret, sourceSecret.getMetadata()
+                    .getNamespace(), sourceSecret.getMetadata().getName());
+
+        }
+        return null;
+    }
+
+    public static synchronized void deleteSourceCredentials(
+            BuildConfig buildConfig) throws IOException {
+        Secret sourceSecret = getSourceCredentials(buildConfig);
+        if (sourceSecret != null) {
+            String labelValue = sourceSecret.getMetadata().getLabels()
+                    .get(Constants.LABEL_JENKINS);
+            boolean watching = labelValue != null
+                    && labelValue.equals(Constants.VALUE_SYNC);
+            // for a bc delete, if we are watching this secret, do not delete
+            // credential until
+            // secret is actually deleted
+            if (watching)
+                return;
+            deleteCredential(sourceSecret);
+        }
+    }
+
+    public static synchronized void deleteCredentials(Secret secret)
```

Comment:
not clear to me why this function exists (vs callers just calling `deleteCredential` directly)

Reply:
meant to consolidate and forgot .... will push momentarily